platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/AppCommunicationComponentWiring.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring.components;
+
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.platform.components.appcomm.AppCommunicationComponent;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Wiring for the {@link AppCommunicationComponent}.
+ *
+ * @param appCommunicationComponentInputWire  the input wire for app communication component
+ */
+public record AppCommunicationComponentWiring(
```

Comment:
There should be another input wire here that accepts a `StateSavingResult`, and is bound to `AppCommunicationComponent::stateSavedToDisk`

Reply:
@lpetrovic05 I think you had a slightly different proposal of putting this in a lambda so caller just calls the lambda instead? Is that functionally different from @alittley's proposal here or are they equivalent?

Reply:
moved stateSavedToDisk as a lambda in SwirldsPlatform.